### ISSUE TITLE
Linter config idea: Allow newline after property colon

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#428]
Some organizations have a character limit for their SCSS lines. The only way I've seen this actually become a problem is with long property values, like URLs. So you might want to allow SCSS like:

```css
body {
  background-image:
      url('some-crazy-long-url-that-doesnt-fit-within-eighty-characters');
}
```

It would be nice to have a config style for SpaceAfterPropertyColon that would allow one space or a newline-and-spaces, but only if the `name: value;` is longer than _x_ characters.

I don't have a great idea what this style config should be called, or how to specify _x_, but maybe someone else has an idea... Maybe `style: one_space_or_newline_over_80_characters`, and the 80 is dynamically parsed out :worried: bleh.

**Comment by sds** [comment]
I'm not sure the validation of the length of the property value is really necessary. I think simply allowing the user to allow newlines will get the job done (how often are you placing property values on their own lines anyway?).

If you really want the enforcement of allowing newlines only for property values over a certain length, you could codify the logic so that specifying an `allow_newline_when_line_exceeds_length: 80` would enforce allowing of newlines only if a number was set. Specifying `0` could mean you allow newlines at any length.

A pull request would be welcome!

**Comment by srawlins** [comment]
Ah, good idea. I think I like the first suggestion. So in addition to `one_space`, `no_space`, `at_least_one_space`, and `aligned`, we could add:

```yaml
linters:
  SpaceAfterPropertyColon:
    style: one_space_or_newline
```

**Comment by sds** [comment]
Implemented by #441. Thanks!
